services/motion/builtin/builtin_test.go

```diff
@@ -193,8 +193,8 @@ func TestMoveSingleComponent(t *testing.T) {
 		worldState := &referenceframe.WorldState{Transforms: transforms}
 
 		poseToGrab := spatialmath.NewPose(
-			r3.Vector{X: -20., Y: 0., Z: -800.},
-			&spatialmath.R4AA{Theta: math.Pi / 2, RX: 1., RY: 0., RZ: 0.},
+			r3.Vector{X: -790., Y: -232., Z: 60.},
```

Comment:
Just so this is out in the open, was there any reason these values needed to change? Feasibility for the `ur5e`?

Reply:
using 
```
r3.Vector{X: -20., Y: 0., Z: -800.},
&spatialmath.R4AA{Theta: math.Pi / 2, RX: 1., RY: 0., RZ: 0.},
```
will cause us to time out.

running `go test -run TestMoveSingle/succeeds_with -v`, this is the terminal output I get for using the "old" pose:
```
logger.go:130: 2023-03-08T19:30:13.684-0500 DEBUG   motionplan/motionPlanner.go:199 motion config for this step: map[motion_profile:linear]
    logger.go:130: 2023-03-08T19:30:13.686-0500 DEBUG   motionplan/combinedInverseKinematics.go:65      starting inputs: [{0} {0} {0} {0} {0} {0}]
    logger.go:130: 2023-03-08T19:30:13.686-0500 DEBUG   motionplan/combinedInverseKinematics.go:71      starting pose: x:-817.2  y:-232.9  z:62.8  o_x:1.5700924586837752e-16  o_y:-1  o_z:1.7938038903913487e-16  theta:89.99999999999999
    logger.go:130: 2023-03-08T19:30:13.686-0500 DEBUG   motionplan/combinedInverseKinematics.go:72      goal pose: x:-798.9999999999999  y:2.0000000000000195  z:22.999999999999844  o_x:2.220446049250313e-16  o_y:-1  o_z:1.232595164407831e-32
    logger.go:130: 2023-03-08T19:30:13.825-0500 DEBUG   motionplan/combinedInverseKinematics.go:65      starting inputs: [{0} {0} {0} {0} {0} {0}]
    logger.go:130: 2023-03-08T19:30:13.825-0500 DEBUG   motionplan/combinedInverseKinematics.go:71      starting pose: x:-817.2  y:-232.9  z:62.8  o_x:1.5700924586837752e-16  o_y:-1  o_z:1.7938038903913487e-16  theta:89.99999999999999
    logger.go:130: 2023-03-08T19:30:13.825-0500 DEBUG   motionplan/combinedInverseKinematics.go:72      goal pose: x:-816.4416666666668  y:-223.11250000000004  z:61.14166666666667  o_x:1.6047275571679382e-16  o_y:-1.0000000000000002  o_z:1.7520988960184599e-16  theta:86.24999999999999
    logger.go:130: 2023-03-08T19:30:14.012-0500 DEBUG   motionplan/cBiRRT.go:180        running CBiRRT with start map of size 1 and goal map of size 11
```

what stands out to me is that for `motionplan/combinedInverseKinematics.go` lines 71-72 get called twice.
The first time line 72 is reached we get:
```
goal pose: x:-798.9999999999999  y:2.0000000000000195  z:22.999999999999844  o_x:2.220446049250313e-16  o_y:-1  o_z:1.232595164407831e-32
```

while the second time, it is:
```
goal pose: x:-816.4416666666668  y:-223.11250000000004  z:61.14166666666667  o_x:1.6047275571679382e-16  o_y:-1.0000000000000002  o_z:1.7520988960184599e-16  theta:86.24999999999999
```

I can see how the first goal pose is difficult to solve for and would cause a time-out.
I am confused why this is happening so tmr I will investigate that.


The use of the updated pose allows for CBiRRT to solve much faster - sans timeout.

Reply:
This is expected behavior.
The first time we reach line 72 is it because we have spawned a solver which checks for feasibility.
The second time we reach line 72 we are actually solving.

I was able to figure this out with help from @raybjork.

perhaps we can start a team conversation about how to structure our logging statements so they provide both high and low level implementation details since the current logs do not relay this information succinctly or in a user friendly manner.